components/Typography/src/UIFont+MaterialScalable.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <UIKit/UIKit.h>
+
+#import "MDCFontTextStyle.h"
+
+@interface UIFont (MaterialScalable)
+
+
+/**
+ An associated scaling curve that is a dictionary that maps UISizeCategory to fontsize
+ */
+@property(nonatomic, nullable, setter=mdc_setScalingCurve:)
+    NSDictionary<UIContentSizeCategory, NSNumber *> *mdc_scalingCurve;
+
+/**
+ Is there a scaling curve associated with the font
+ */
+@property(nonatomic, readonly, getter=mdc_hasScalingCurve) BOOL mdc_scalingCurveAssociated;
```

Comment:
Do we need this API? In other words: can a client rely on a nil return value from mdc_scalingCurve to indicate the same thing?

Reply:
This is a convenience method.  

To be clear, the underlying implementation is exactly what you mention above, there is no associated storage:
```
- (BOOL)mdc_hasScalingCurve {
  if (self.mdc_scalingCurve != nil) {
    return YES;
  } else {
    return NO;
  }
}
```

During discussions with clients, some wanted to use a typography scale without scalable fonts to indicate that a dynamic type should not be enabled.  They will end up querying a UIFont for it's scalingCurve a decent amount in their code.  It's the future readability of:

if (font.mdc_hasScalingCurve)
if (font.mdc_scalingCurve != nil)

I'm happy to remove if you feel strongly.

Reply:
I'm not strongly opposed to this convenience accessor, but I feel like this:
`if (font. mdc_hasScalingCurve)`
isn't actually any more convenient than:
`if (font.mdc_scalingCurve)`